ambry-account/src/main/java/com/github/ambry/account/mysql/MySqlAccountStoreFactory.java

```diff
@@ -25,43 +27,42 @@
 
 
 /**
- * Factory class to return an instance of {@link MySqlAccountStore} on {@link #getMySqlAccountStore(boolean)} call
+ * Factory class to return an instance of {@link MySqlAccountStore} on {@link #getMySqlAccountStore()} call
  */
 public class MySqlAccountStoreFactory {
 
   private static final Logger logger = LoggerFactory.getLogger(MySqlAccountStoreFactory.class);
+  private static final String UNKNOWN_DATACENTER = "Unknown";
+
   protected final MySqlAccountServiceConfig accountServiceConfig;
+  protected final String localDatacenter;
 
   /**
    * Constructor.
-   * @param  verifiableProperties The properties to get a {@link MySqlAccountStore} instance. Cannot be {@code null}.
+   * @param verifiableProperties The properties to get a {@link MySqlAccountStore} instance. Cannot be {@code null}.
    */
   public MySqlAccountStoreFactory(VerifiableProperties verifiableProperties) {
     this.accountServiceConfig = new MySqlAccountServiceConfig(verifiableProperties);
+    // If datacenter is unknown, all endpoints will be considered remote
+    this.localDatacenter =
+        verifiableProperties.getString(ClusterMapConfig.CLUSTERMAP_DATACENTER_NAME, UNKNOWN_DATACENTER);
   }
 
   /**
    * Returns an instance of the {@link MySqlAccountStore} that the factory generates.
-   * @param writeable needs to be set to true if requesting a store that accepts writes.
    * @return an instance of {@link MySqlAccountStore} generated by this factory.
    * @throws SQLException
    */
-  public MySqlAccountStore getMySqlAccountStore(boolean writeable) throws SQLException {
+  public MySqlAccountStore getMySqlAccountStore() throws SQLException {
     Map<String, List<MySqlUtils.DbEndpoint>> dcToMySqlDBEndpoints = getDbEndpointsPerDC(accountServiceConfig.dbInfo);
-    for (List<MySqlUtils.DbEndpoint> dbEndpoints : dcToMySqlDBEndpoints.values()) {
-      // TODO: Can have logic to try mysql end point on local DC first
-      for (MySqlUtils.DbEndpoint dbEndpoint : dbEndpoints) {
-        if ((dbEndpoint.isWriteable() && writeable) || (!dbEndpoint.isWriteable() && !writeable)) {
-          try {
-            return new MySqlAccountStore(dbEndpoint);
-          } catch (SQLException e) {
-            logger.error("MySQL account store creation failed", e);
-            // TODO: Add logic to retry on different db end point based on type of exception.
-            throw e;
-          }
-        }
-      }
+    // Flatten to List (TODO: does utility method need to return map?)
```

Comment:
I was gonna reuse `getDbEndpointsPerDC` in some code of mine too, but it does not really need to return a map in my case either.